plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -563,6 +668,74 @@ func (s *SQLServer) refreshToken() (*adal.Token, error) {
 	return s.adalToken, nil
 }
 
+//------------------------------------------------------------------------------
+// New Azure Identity SDK Token Methods
+//------------------------------------------------------------------------------
+
+// loadAzureToken loads a token from in-memory cache using the Azure Identity SDK.
+//
+// This is the recommended authentication method for Azure SQL resources.
+func (s *SQLServer) loadAzureToken() (*azureToken, error) {
+	// This method reads from variable (in-mem cache) but can be extended
+	// for different cache mechanisms in the future
+
+	if s.azToken == nil {
+		return nil, errors.New("token is nil or failed to load existing token")
+	}
+
+	return s.azToken, nil
+}
+
+// refreshAzureToken refreshes the token using the Azure Identity SDK.
+//
+// This is the recommended authentication method for Azure SQL resources.
+func (s *SQLServer) refreshAzureToken() (*azureToken, error) {
+	var cred azcore.TokenCredential
+	var err error
+
+	if s.ClientID == "" {
+		// Use system-assigned managed identity
+		s.Log.Debugf("Using system-assigned managed identity with Azure Identity SDK")
+		cred, err = azidentity.NewManagedIdentityCredential(nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create managed identity credential (system-assigned): %w", err)
+		}
+	} else {
+		// Use user-assigned managed identity
+		s.Log.Debugf("Using user-assigned managed identity with ClientID: %s with Azure Identity SDK", s.ClientID)
+		options := &azidentity.ManagedIdentityCredentialOptions{
+			ID: azidentity.ResourceID(s.ClientID),
+		}
+		cred, err = azidentity.NewManagedIdentityCredential(options)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create user-assigned managed identity credential: %w", err)
+		}
+	}
```

Comment:
Why not doing
```suggestion
	var options *ManagedIdentityCredentialOptions
	if s.ClientID != "" {
		options = &azidentity.ManagedIdentityCredentialOptions{
			ID: azidentity.ResourceID(s.ClientID),
		}
	}

	cred, err := azidentity.NewManagedIdentityCredential(options)
	if err != nil {
		return nil, fmt.Errorf("failed to create managed identity credential (system-assigned): %w", err)
	}
```

Reply:
To be honest I would just remove the `(system-assigned)` part of the error message. Please also keep in mind that many people do enable debugging to see the buffer fullness in log files so I won't add these messages...